pkg/neonvm/controllers/vmmigration_controller.go

```diff
@@ -752,3 +772,55 @@ func (r *VirtualMachineMigrationReconciler) targetPodForVirtualMachine(
 
 	return pod, nil
 }
+
+// hasArchitectureAffinity checks if the VM has the affinity with the CPU architecture of the source node
+func hasArchitectureAffinity(vm *vmv1.VirtualMachine, sourceNode *corev1.Node) bool {
+	if vm.Spec.Affinity != nil && vm.Spec.Affinity.NodeAffinity != nil && vm.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+		for _, term := range vm.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms {
+			for _, expr := range term.MatchExpressions {
+				if expr.Key == "kubernetes.io/arch" && expr.Operator == corev1.NodeSelectorOpIn && len(expr.Values) > 0 {
+					for _, value := range expr.Values {
+						if value == sourceNode.Status.NodeInfo.Architecture {
+							return true
+						}
+					}
+				}
+			}
+		}
+	}
+	return false
+}
```

Comment:
This function can be rewritten to reduce the max nested level from 7 to 4, smth like below.

This implementation also doesn't use sourceNode.
```suggestion
func hasArchitectureAffinity(vm *vmv1.VirtualMachine, sourceNode *corev1.Node) bool {
	if vm.Spec.Affinity == nil || vm.Spec.Affinity.NodeAffinity == nil || vm.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
		return false
	}
	
	terms = vm.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
	for _, term := range terms {
		for _, expr := range term.MatchExpressions {
			if expr.Key == "kubernetes.io/arch" && 
			   expr.Operator == corev1.NodeSelectorOpIn && 
			   len(expr.Values) == 1 {
				return true		
			}
		}
	}
	
	return false
}
```